### PR TITLE
KT-40235 Fix external interface companion support

### PR DESCRIPTION
**Pull request by turansky**
Related issues:
* [KT-40235](https://youtrack.jetbrains.com/issue/KT-40235) - IR. Broken companion support
* [KT-44394](https://youtrack.jetbrains.com/issue/KT-44394) - IR. Broken external interfaces `NodeFilter` and `SVGUnitTypes`
* [KT-39602](https://youtrack.jetbrains.com/issue/KT-39602) - Companion legalize

cc @romanart

### CODE REVIEW

**Comment by aerialist7** [comment]
Cool, we need it too

**Comment by romanart** [comment]
Would be great if you add some test for it

**Review by skuzmich — CHANGES_REQUESTED** [review]
This PR removes support for string union hack we have in stdlib. For example:

https://github.com/JetBrains/kotlin/blob/3ad597ca5c704a48092a4969daea41021cbd5db9/libraries/stdlib/js/src/org.w3c/org.w3c.dom.mediasource.kt#L79-L89

Nobody likes this hack, but we have to support it until we replace it with something else. This is why I cannot approve this PR as it is.

Temporary workaround would be to apply this hack only to companions from stdlib. For example, by annotating them with some marker internal annotation or by checking package names.

Compiler test this covers the code that this PR removes:
js/js.translator/testData/box/regression/companionObjectInExternalInterface.kt

File: libraries/stdlib/js/src/org.w3c/org.w3c.dom.encryptedmedia.kt
Comment (skuzmich) [inline libraries/stdlib/js/src/org.w3c/org.w3c.dom.encryptedmedia.kt:181]: This is a clever hack! But these files are autogenerated and fix should land in dukat. @Schahen could you help?

**Review by skuzmich — CHANGES_REQUESTED** [review]

**Review by Schahen — CHANGES_REQUESTED** [review]
Victor, this indeed makes sense, the only reason this can be adopted is that actually this part of stdlib is generated from the webidl source set, so I suggest you to choose one of two options:

1.  Create similar PR in dukat  - https://github.com/Kotlin/dukat - this is the tool that is used for generating sources from idl - something like if there's an interface with string union - add JsName("null")
2.  Leave it to us till we introduce such changes.

Of course, both options are totally fine - obviously not everyone has time to contribute to any open-source project.

Will you choose option #1 though,  you can count on our full cooperation and feel free to reach with any question you have personally me, either via email or in Slack.

**Comment by Schahen** [comment]
I've just realised that this particular PR includes some other changes as well (not related to changes in generated code) so I was wrong closing it.

**Comment by turansky** [comment]
Dukat PR is ready - https://github.com/Kotlin/dukat/pull/430

File: libraries/stdlib/js/src/org.w3c/org.w3c.dom.encryptedmedia.kt
Comment (turansky) [reply libraries/stdlib/js/src/org.w3c/org.w3c.dom.encryptedmedia.kt:184]: Fixed. Declaration regenerated by Dukat `0.5.8-rc.4` (new)

**Comment by skuzmich** [comment]
Thanks! Pushed this PR manually with a couple of additional fixes.

**Pull request closed without merging**
